### PR TITLE
SNOW-1653121: Support some Timedelta aggregations on axis=0.

### PR DESCRIPTION
**Pull request by sfc-gh-mvashishtha**
Fixes SNOW-1653121

Test and support aggregation on axis=0. We still raise `NotImplementedError` if:

1) the aggregation requires concatenating a frame with timedelta types
2) the aggregation requires transposing a row containing a timedelta type and other types.

To facilitate this change, introduce the class `SnowparkPandasAggFunc`, which represents a Snowpark pandas aggregation, together with the bool `preserves_snowpark_pandas_type`. We can use `SnowparkPandasAggFunc` instead of the separate mappings `GROUPBY_AGG_PRESERVES_SNOWPARK_PANDAS_TYPE` and `GROUPBY_AGG_WITH_NONE_SNOWPARK_PANDAS_TYPES`.

This change also fixes the bug that timedelta aggregations like mean would produce the wrong type (and the wrong result) by truncating the float result if `preserves_snowpark_pandas_type`.

### CODE REVIEW

**Comment by sfc-gh-mvashishtha** [comment]
marking as draft till I fix SNOW-1654730

**Review by sfc-gh-nkrishna — APPROVED** [review]
Nice work, LGTM!

File: tests/integ/modin/conftest.py [hunk 1]
```diff
@@ -715,3 +715,29 @@ def numeric_test_data_4x4():
...
+                pd.Timedelta(minutes=-1),
+                pd.Timedelta(minutes=0),
+                pd.Timedelta(minutes=5),
+                pd.Timedelta(minutes=6),
+            ],
+            "C": [
+                None,
+                pd.Timedelta(nanoseconds=5),
+                pd.Timedelta(nanoseconds=0),
+                pd.Timedelta(nanoseconds=4),
```
Comment (sfc-gh-nkumar) [inline tests/integ/modin/conftest.py:740]: not sure if this covered in other tests. It would good to include a column withe all None values.

**Review by sfc-gh-nkumar — APPROVED** [review]

File: tests/integ/modin/conftest.py (hunk 1 above)
Comment (sfc-gh-mvashishtha) [reply tests/integ/modin/conftest.py:740]: Done.

**Pull request merged**